docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Implemented Features
 
+Following a restucture to the project to follow a more Object Oriented structure, as well as to allow for conformity and easy editing of code, many of our commands will follow a similar structure of the following:
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user (output for the user is stored in a `Response` class)
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
+
+**Step1.** After recieving the necessary method call, the `CommandHandler` class will self-call executeCommand with both the list of projects as well as a constructor call to AddProjectCommandParser (and a call to the parse method of AddProjectCommandParser) as its parameters
+
+**Step2.** The AddProjectCommandParser object instance is created
+
+**Step3.** The parse method of AddProjectCommandParser is called with `commands` (the user input) as its parameter.
+
+**Step4.** AddProjectCommandParser will self call its `getProjectTitle` method. This will split the user's input and return the correct name for the project
+
+**Step5.** After returning this projectTitle to itself, `AddProjectCommandParser` will call the constructor for a new `AddProjectCommand` object
+
+**Step6.** This `AddProjectCommand` object will use the newly gathered project name to call the `addProjct` command of the `ProjectList` class.
+
+**Step7.** The status of this command completing is returned to `CommandHandler`, and the output is sent to the user.
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
```

Comment:
1. The `deadline` returned from `Project` object should perhaps be at the end of the activation bar
![image](https://user-images.githubusercontent.com/69310031/160963463-70ecc875-4fe7-44a8-9e59-9b3ff8dcb556.png)

2. It would be good to ensure that all the arrows (that are relevant) start and end at the edge of the activation bars
![image](https://user-images.githubusercontent.com/69310031/160963488-3df4b62a-9aff-4c8e-a2b8-f54d41f090e7.png)


3. Should there be a return from self-invocation of `CommandHandler`? (Not compulsory but just in case it returns something or so)
![image](https://user-images.githubusercontent.com/69310031/160963517-a57c0d74-9818-4fc6-9a9e-79623ed3c14b.png)